index.js

```diff
@@ -212,6 +212,17 @@
     function(x) { return x != null; }
   );
 
+  //  Applicative :: TypeClass
+  var Applicative = $.TypeClass(
+    'sanctuary/Applicative',
+    function(x) {
+      return (
+        _type(x) === 'Array' ||
+        Apply.test(x) && (hasMethod('of')(x) || hasMethod('of')(x.constructor))
+      );
+    }
+  );
+
```

Comment:
> It might be worthwhile adding some functionality to `$.TypeClass` to help extending type classes like in this example.

Good idea. Do you have an API in mind? Here are two options:

```javascript
//    Applicative :: TypeClass
const Applicative = $.TypeClass(
  'sanctuary/Applicative',
  $.extends(
    [Apply],
    x => _type(x) === 'Array' || hasMethod('of')(x) || hasMethod('of')(x.constructor)
  )
);
```

```javascript
//    Applicative :: TypeClass
const Applicative = $.TypeClass(
  'sanctuary/Applicative',
  [Apply],
  x => _type(x) === 'Array' || hasMethod('of')(x) || hasMethod('of')(x.constructor)
);
```